### PR TITLE
feat: support mode parameter in defineConfig

### PR DESCRIPTION
**Pull request by Jungzl**
## Summary

support mode parameter in `defineConfig`

## Checklist

- [ ] Tests updated.
- [x] Documentation updated.

### CODE REVIEW

File: packages/core/src/cli/commands.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export async function init({
     const { content: config, filePath: configFilePath } = await loadConfigV2({
       cwd: root,
       path: commonOpts.config,
+      mode: commonOpts.envMode,
```
Comment (chenjiahan) [inline packages/core/src/cli/commands.ts:59]: The mode and envMode are two different concept, so we want to distinguish between these two concepts.

- **mode**: currently uses NODE_ENV, equivalent to webpack’s mode concept, basically used to set some build configurations for a specific environment.

- **env mode**: can be set via `--env-mode <value>`, only be used to load `.env.[env-mode]` file. Can be set to any value.

Comment (chenjiahan) [reply packages/core/src/cli/commands.ts:59]: At the same time, I am worried that exposing envMode in rsbuild.config.ts is a potential burden for users who use the Rsbuild JavaScript API. 

They also need to pass in this value to `loadConfig` method to ensure that rsbuild.config.ts can get this value.

Comment (Jungzl) [reply packages/core/src/cli/commands.ts:59]: Got it, what I want to add is `envMode`

File: packages/core/src/cli/config.ts [hunk 2]
```diff
@@ -115,6 +116,7 @@ export async function loadConfigByPath(configFile: string) {
       const params: ConfigParams = {
         env: getNodeEnv(),
         command,
+        envMode,
```
Comment (chenjiahan) [inline packages/core/src/cli/config.ts:119]: should we use `getNodeEnv()` as the default value of `envMode`?

Adding default values can be consistent with the behavior of `loadEnv` method

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
👍👍

**Pull request merged**